skimage/segmentation/tests/test_random_walker.py

```diff
@@ -94,11 +94,15 @@ def test_2d_cg(dtype):
     ly = 100
     data, labels = make_2d_syntheticdata(lx, ly)
     data = data.astype(dtype, copy=False)
-    with expected_warnings(['"cg" mode|scipy.sparse.linalg.cg']):
+    with expected_warnings(
+        ['Changing the sparsity structure|"cg" mode|scipy.sparse.linalg.cg']
+    ):
```

Comment:
Might be worth updating to `pytest.warns` here? It can be a lot more explicit (warning type), and allow for follow-up inspection. I've been trying to reduce our reliance on custom non-standard test tooling.

Also it seems dubious to me that the following line of code might raise three different warnings that need to be "or"-ed? I suspect that it is always the same one warning? Is it possible to make this more explicit?

Reply:
Feel free to push to the PR; Looks like this is simply a reformatting due to ruff.

Reply:
We can do that in another PR as long as we don't forget. ;)